Hybrid Link API

nemijato opened the pull request:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

nemijato committed:
Add swagger changes regarding initiating hybrid link create operation.

AutorestCI commented:
You don't have permission to trigger SDK Automation.
Please add yourself to ***Azure*** group from opensource portal if you are MSFT employee,
or please ask reviewer to add comment *** /openapibot sdkautomation ***.
Please ask tih@microsoft.com (or NullMDR in github) for additional help.

nemijato committed:
Add new line at the end of the json files.

azuresdkci commented:
Can one of the admins verify this patch?

ravbhatnagar commented on specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/hybridLinkCreate.json:
```diff
@@ -0,0 +1,280 @@
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/managedInstances/{managedInstanceName}/hybridLinkCreate": {
```
creating something via a POST is not recommended in Azure. It cant be invoked via ARM template, Azure policy doesnt work against POST, its not idempotent etc. Can this be exposed via a PUT?

ravbhatnagar replied:
and once you add the PUT, you will have to support GET, DELETE etc. which will enable the complete management experience for this. otherwise how does someone get this once it is created? 
another question, is this a singleton? can only one hybridLInk be created under a managedINstance?

nemijato replied:
It can be exposed via PUT I guess.
We decided to implement delete support a bit later, as our feature will not support it in preview.
I'll implement a GET and it is not a singleton.

Btw thanks for the comments.

ravbhatnagar reviewed (commented):
@nemijato couple of comments

chiragg4u commented:
Please go through the contribution checklist added as part of the review and mark them completed. This is a requirement before review can be continued.

ravbhatnagar commented:
@nemijato - are you planning to change the POST to PUT? and the other recommended changes?

nemijato committed:
Implement put, get, list and delete requests and add examples.

nemijato commented:
I run oav validate-spec and oav validate-examples and fixed all errors.
One error I didn't fix is from the ValidateSwagger command:
(OperationsAPIImplementation/R3023/ARMViolation): Operations API must be implemented for '/providers//operations'.
    - file:///E:/DsMainDev/obj/x64debug/Sql/xdb/manifest/svc/mgmt/WebApiHosting/ARM/tools/arm.tools.csproj/swagger/Microsoft.Sql/2020-02-02-preview/swagger.json:17:2

I didn't fix it as I implemented both operation results and azure async operation result get requests.

nemijato committed:
Fix avocado, SpellCheck and probably some other checks

nemijato committed:
Fix lint error where delete example doesn't contain response 204.

nemijato committed:
Add comma

nemijato committed:
Add delete response 204 into swagger

nemijato committed:
Add secondary AG parameter to put request.

nemijato commented on specification/sql/resource-manager/Microsoft.Sql/preview/2020-02-02-preview/examples/HybridLinkUpdate.json:
```diff
@@ -0,0 +1,27 @@
+    "subscriptionId": "00000000-1111-2222-3333-444444444444",
+    "resourceGroupName": "testrg",
+    "managedInstanceName": "testcl",
+    "dagName": "dag",
```
distributedAvailabilityGroupName is the acronym. If you don't think its too long I'll expand it.

nemijato committed:
Regenerate swagger and adjust examples based on last ARM changes

nemijato committed:
Update swagger and examples

nemijato committed:
Change resource name description

akning-ms commented:
@ericshape  @nemijato  as PR#11840  (version 2020-02-02-preview) already merged. this PR is not required any more, right? if yes. can you help close it?
